am-lib/src/main/java/uk/gov/hmcts/reform/amlib/AccessManagementService.java

```diff
@@ -263,6 +246,46 @@ public FilteredResourceEnvelope filterResource(@NotBlank String userId,
         return permissionsService.merge(permissionsForRoles);
     }
 
+    /**
+     * Retrieves {@link RolePermissions} filtered by role security classification for a specific resource and role.
+     *
+     * @param resourceDefinition {@link ResourceDefinition} a unique service name, resource type and resource name
+     * @param roleName           user role name
+     * @return {@link RolePermissions} a combination of permissions and security classifications for a role name
+     * @throws PersistenceException if any persistence errors were encountered
+     */
+    @AuditLog("returned role access to resource defined as '{{resourceDefinition.serviceName}}|"
+        + "{{resourceDefinition.resourceType}}|{{resourceDefinition.resourceName}}' for role '{{roleName}}': "
+        + "{{result}}")
+    public RolePermissions getRolePermissions(@NotNull @Valid ResourceDefinition resourceDefinition,
+                                              @NotBlank String roleName) {
+        Map<AccessType, SecurityClassification> roleData = jdbi.withExtension(AccessManagementRepository.class, dao ->
+            dao.getRoles(Collections.singleton(roleName), Stream.of(EXPLICIT, ROLE_BASED).collect(toSet())))
+            .stream().collect(toMap(Role::getAccessType, Role::getSecurityClassification));
+
+        if (roleData.isEmpty()) {
+            return null;
+        }
+
+        List<AttributeData> attributeData = jdbi.withExtension(AccessManagementRepository.class, dao ->
+            dao.getAttributeSecurityClassificationsForResource(
+                resourceDefinition, roleName, SecurityClassifications.fromValueOf(
+                    roleData.entrySet().iterator().next().getValue().getHierarchy())));
+
+        if (attributeData.isEmpty()) {
+            return null;
+        }
+
+        return RolePermissions.builder()
+            .accessType(roleData.entrySet().iterator().next().getKey())
```

Comment:
A sequence of `roleData.entrySet().iterator().next()` appears quite often, shall we extract both role's access type and default security classification as local variables?

Reply:
sure

Reply:
access type doesn't make sense, only used once. Sec classification, sure